### PR TITLE
Server refactoring

### PR DESCRIPTION
**Pull request by veloman-yunkan**
This refactoring was performed in the course of the work on kiwix/kiwix-tools#140, a significant part of which is now complete under #340. However that PR started growing too large, and I decided to break it into several smaller ones. This PR is the second one of them, preceded by #344 (which provides additional protection against this PR introducing some bugs).

As a series of relatively small refactoring changes, this PR better be reviewed one commit at a time.

### CODE REVIEW

**Comment by veloman-yunkan** [comment]
@rgaudin Sorry. I included you because that's what GitHub suggested.

File: src/server.cpp [hunk 1]
```diff
@@ -907,27 +903,32 @@ Response InternalServer::handle_content(const RequestContext& request)
...
     if (m_verbose.load())
       printf("Failed to find %s\n", urlStr.c_str());
 
     return build_404(request, bookName);
   }
 
+  if (entry.isRedirect() || urlStr.empty()) {
+    // If urlStr is empty, we want to mainPage.
+    // We must do a redirection to the real page.
+    return build_redirect(bookName, entry.getFinalEntry());
```
[the remark this answers is not in the document]
Comment (veloman-yunkan) [reply src/server.cpp]: I try, you catch,
Here is a new patch:

https://github.com/kiwix/kiwix-lib/pull/345/commits/6c7ab6ff54e67c9c23c2b271caaf4b7e99934bf5

:)

**Comment by veloman-yunkan** [comment]
> There is also very few changes made in a wrong commit (remove unused variable in one commit after you remove the usage of this variable in previous commit).

I had fixed at least one such situation while extracting this PR out of #340, but obviously missed something. Chasing those instances that escaped that step isn't justified now. After all, the true result of this refactoring is the whole sequence rather than each one of them separately, even though they **aren't** going to be squashed (I hope so).

**Comment by mgautierfr** [comment]
> I had fixed at least one such situation while extracting this PR out of #340, but obviously missed something. Chasing those instances that escaped that step isn't justified now. After all, the true result of this refactoring is the whole sequence rather than each one of them separately, even though they aren't going to be squashed (I hope so).

I totally agree. And the commits will not be squashed, having a nice git history is really useful. That is why I'm sometime a bit pedantic on the git history of the PR.

**Pull request merged**